### PR TITLE
COMPASS-4077 - Notarize builds for macOS

### PR DESCRIPTION
**Pull request by addaleax**
- https://github.com/mongodb-js/electron-installer-codesign/pull/6
- https://github.com/mongodb-js/hadron-build/pull/365
- https://github.com/mongodb-js/electron-installer-codesign/pull/7
- https://github.com/mongodb-js/hadron-build/pull/367
- https://github.com/mongodb-js/hadron-build/pull/368

🤞

### Checklist
- [ ] New tests and/or benchmarks are included
- [ ] Documentation is changed or added

## Motivation and Context

- [ ] Bugfix
- [ ] New feature
- [x] Dependency update
- [x] Misc

## Types of changes

- [ ] *Backport Needed*
- [x] Patch (non-breaking change which fixes an issue)
- [ ] Minor (non-breaking change which adds functionality)
- [ ] Major (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -384,7 +384,7 @@
     "enzyme": "^3.11.0",
     "enzyme-adapter-react-16": "^1.15.2",
     "eslint-config-mongodb-js": "^5.0.3",
-    "hadron-build": "^23.2.6",
+    "hadron-build": "^23.3.0",
```
Comment (mcasimir) [inline package.json:387]: `23.3.1` :)

Comment (addaleax) [reply package.json:387]: I’ve updated to that – I have no strong feelings about whether we want to drop the write-to-file step then, after sleeping about it it seems more likely to me that somebody would accidentally print environment variables like we did before than accidentally publish files from `/tmp/`, but obviously either way works just fine for me.

**Comment by Anemy** [comment]
Is there a case where we don't want to notarize the bundle? Not really related to this PR, but in https://github.com/mongodb-js/hadron-build/pull/367/files#diff-2265b5363474c70d8268d7c4f5e1e060L545 when the apple credentials aren't set we still continue with the build, I'm wondering if we want to have it error out when the credentials don't exist to make sure we don't release without notarizing.

**Comment by addaleax** [comment]
@rhys To be honest, I don’t know, and I think it might just be that that was the case because we weren’t able to successfully notarize in the past. I don’t see any downsides to doing it always, though, and we also notarize every `mongosh` build.

**Pull request merged**